File: Algorithm/Implementation/The Hurdle Race/The Hurdle Race.py
#!/bin/python3

import sys

def hurdleRace(k, height):
    # Complete this function
    '''
	# method 1
    maxheight = sorted(height)[-1]
    return maxheight - k if( maxheight > k) else 0
    '''
	
	# method 2
    return max( max( height) - k, 0)

if __name__ == "__main__":
    n, k = input().strip().split(' ')
    n, k = [int(n), int(k)]
    height = list(map(int, input().strip().split(' ')))
    result = hurdleRace(k, height)
    print(result)

File: Algorithm/Implementation/The Hurdle Race/The Hurdle Race.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

int hurdleRace(int k, int height_size, int* height) {
    // Complete this function
    int max = height[0];
    for(int i = 1; i < height_size; i++){
        if(max < height[i])
            max = height[i];
    }
    return (k > max) ? 0 : (max - k);
}

int main() {
    int n; 
    int k; 
    scanf("%i %i", &n, &k);
    int *height = malloc(sizeof(int) * n);
    for (int height_i = 0; height_i < n; height_i++) {
       scanf("%i",&height[height_i]);
    }
    int result = hurdleRace(k, n, height);
    printf("%d\n", result);
    return 0;
}
